scripts/quickbar.lua

```diff
@@ -67,7 +67,7 @@ function mod.set_quick_bar_slot(index, pindex)
    local page = game.get_player(pindex).get_active_quick_bar_page(1) - 1
    local stack_cur = game.get_player(pindex).cursor_stack
    local stack_inv = players[pindex].inventory.lua_inventory[players[pindex].inventory.index]
-   local ent = get_selected_ent(pindex)
+   local ent = pop_ent_from_tile(pindex)
```

Comment:
So if we have a new iterator that does what we originally wanted from the entity getting function, I suppose i can switch all these other calls to the new function. They would have to be individually tested but it would make everything more robust.